### PR TITLE
Add operator deploy e2e test

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind design

#### What this PR does / why we need it:
The e2e test will be the base for further enhance tests, right now we
just build the container image and deploy the operator into the kind
cluster.
#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,13 +27,6 @@ RUN adduser \
     --uid "${UID}" \
     "${USER}"
 
-# Speed up build by leveraging docker layer caching
-COPY go.mod go.sum ./
-RUN go mod download
-
-ADD . /work
-RUN make
```
Comment (saschagrunert) [inline Dockerfile:36]: We now build locally, which should be fine since we're building a static binary.

**Comment by saschagrunert** [comment]
Ready to be reviewed, PTAL @hasheddan @pjbgf

File: Dockerfile [hunk 2]
```diff
@@ -28,21 +27,14 @@ RUN adduser \
...
-
-FROM alpine
+FROM scratch
 LABEL name="Seccomp Operator" \
       version="v0.0.1" \
       description="The Seccomp Operator makes it easier for cluster admins to manage their seccomp profiles and apply them to Kubernetes' workloads."
 
 COPY --from=build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
 COPY --from=build /etc/passwd /etc/passwd
```
Comment (pjbgf) [inline Dockerfile:36]: We could follow the same logic and make this simpler by amending the files directly:
```suggestion
RUN echo "2000:x:2000:2000::/nonexistent:/sbin/nologin" >> /etc/passwd &&
    echo "2000:x:2000:" >> /etc/group
```

File: Dockerfile [hunk 3]
```diff
@@ -28,21 +27,14 @@ RUN adduser \
...
 COPY --from=build /etc/group /etc/group
```
Comment (pjbgf) [inline Dockerfile:37]:
```suggestion
```

File: Dockerfile [hunk 4]
```diff
@@ -14,7 +14,6 @@
 
 FROM golang:1.14-alpine AS build
 WORKDIR /work
-RUN apk --no-cache add build-base git gcc
 
 ENV USER=secuser
```
Comment (pjbgf) [inline Dockerfile:18]: Conversely, we could remove the entire first stage image.

**Comment by saschagrunert** [comment]
> If we are to stop using multi-staged images to build the project, we probably should drop the concept altogether and amend `/etc/passwd/` and `/etc/group` directly on the last image.
> 
> I personally prefer the multi-staged approach to build as it decreases the entry effort to contribute to the project. New contributors only need docker installed in their machines. It can also be handy to have the same errors between machines and CI.

Hm okay I revert my changes back and stick to the multi-layer build approach. I think we can leave those parts as they currently are. :+1:

File: Dockerfile (hunk 4 above)
Comment (saschagrunert) [reply Dockerfile:18]: See https://github.com/saschagrunert/seccomp-operator/pull/28#issuecomment-651650217

File: Dockerfile (hunk 3 above)
Comment (saschagrunert) [reply Dockerfile:37]: See https://github.com/saschagrunert/seccomp-operator/pull/28#issuecomment-651650217

File: Dockerfile (hunk 2 above)
Comment (saschagrunert) [reply Dockerfile:36]: See https://github.com/saschagrunert/seccomp-operator/pull/28#issuecomment-651650217

**Comment by saschagrunert** [comment]
PTAL @hasheddan @pjbgf

File: test/suite_test.go [hunk 5]
```diff
@@ -35,50 +52,83 @@ func (e *e2e) SetupSuite() {
...
 				kindVersion+"/kind-linux-amd64",
-		).RunSuccess())
+		)
 		e.Nil(os.Chmod(e.kindPath, 0o755))
 	}
 }
 
 // SetupTest starts a fresh kind cluster for each test
 func (e *e2e) SetupTest() {
+	// Deploy the registry
```
Comment (hasheddan) [inline test/suite_test.go:73]: Could we skip doing this and just load the image into the kind cluster? https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster

Comment (saschagrunert) [reply test/suite_test.go:73]: Hah, yeah I change the implementation as suggested and will choose a pull policy of `Never`.

Comment (hasheddan) [reply test/suite_test.go:73]: This is one of my favorite features of `kind` :)

**Comment by saschagrunert** [comment]
Done changing to pre-load the images instead of deploying a registry. PTAL @hasheddan :)

**Comment by saschagrunert** [comment]
Thank you for the review!

**Pull request merged**